edgarpecero/OMN-Order-Service
# [OMN-31] Feature/implementation of sql server

### Summary of Changes
- Implemented the feature for SQL Server integration, ensuring proper configuration for database connections and data management.

### Task References
- [OMN-X] Implementation of SQL Server as part of database feature expansion.

### Detailed Changes
- Added configurations for SQL Server connection properties.
- Created new classes to handle SQL Server interactions and connections.
- Utilized the Singleton design pattern for database connection pooling to enhance resource management and performance.
- Ensured compatibility and setup for SQL Server within the existing project framework.

### Testing Instructions
1. Verify the SQL Server connection by checking connection properties in the configuration file.
2. Run database-related service methods and confirm successful data retrieval and storage.
3. Test scenarios involving transaction handling to ensure data consistency and integrity.
4. Confirm that error handling is in place for database connection issues.

### Future Considerations
- Extend support for additional SQL Server features, such as stored procedures or triggers, as needed.
- Review and optimize performance for high-volume data transactions.
- Implement automated testing for database interactions.

---

### Additional Notes
- This PR includes initial integration for SQL Server. Future improvements may include additional optimizations and scaling for larger datasets.
